ambry-router/src/main/java/com/github/ambry/router/CompressionMetrics.java

```diff
@@ -0,0 +1,154 @@
+/*
+ * Copyright 2022 LinkedIn Corp. All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ */
+package com.github.ambry.router;
+
+import com.codahale.metrics.Counter;
+import com.codahale.metrics.Histogram;
+import com.codahale.metrics.Meter;
+import com.codahale.metrics.MetricRegistry;
+import com.github.ambry.compression.Compression;
+import java.util.Map;
+import java.util.concurrent.ConcurrentHashMap;
+
+public class CompressionMetrics {
+  public static class AlgorithmMetrics {
+
+    // Algorithm's compress() metrics.
+    public final Meter compressRate;      // Rate compress() is called.
+    public final Counter compressError;   // Rate compress() throws exception.
+    public final Histogram compressRatioPercent;    // compression ratio % = 100 * UncompressedSize/CompressedSize
+    public final Histogram fullSizeCompressTimeInMicroseconds;
```

Comment:
let's simplify the name to TimeInMs.

Reply:
Milliseconds is a big unit.  Typical responsive REST calls take 1-10 ms.  Compression time could take less than 1 ms.  In generally, it's okay to use Milliseconds to measure length operations like external calls, but for internal calls, milliseconds doesn't work unless the value type is a double instead of long.  For this case, it's actually better to use microseconds.

Reply:
If we are ensuring that we will not compress smaller blobs, do we need this metric still?